packages/two_dimensional_scrollables/example/lib/tree_view/custom_tree.dart

```diff
@@ -0,0 +1,275 @@
+            TreeViewNode<String>('table_test.dart'),
+          ],
+        ),
+        TreeViewNode<String>(
+          'tree_view',
+          children: <TreeViewNode<String>>[
+            TreeViewNode<String>('render_tree_test.dart'),
+            TreeViewNode<String>('tree_core_test.dart'),
+            TreeViewNode<String>('tree_delegate_test.dart'),
+            TreeViewNode<String>('tree_span_test.dart'),
+            TreeViewNode<String>('tree_test.dart'),
+          ],
+        ),
+      ],
+    ),
+  ];
+
+  Widget _treeNodeBuilder(
+    BuildContext context,
+    TreeViewNode<dynamic> node, {
+    AnimationStyle? animationStyle,
+  }) {
+    final bool isParentNode = node.children.isNotEmpty;
+    final BorderSide border = BorderSide(
+      width: 2,
+      color: Colors.purple[300]!,
+    );
+    // TRY THIS: TreeView.toggleNodeWith can be wrapped around any Widget (even
+    // the whole row) to trigger parent nodes to toggle opened and closed.
+    // Currently, the toggle is triggered in _getTapRecognizer below using the
+    // TreeViewController.
+    return Row(
+      children: <Widget>[
+        // Custom indentation
+        SizedBox(width: 10.0 * node.depth! + 8.0),
+        DecoratedBox(
+          decoration: BoxDecoration(
+            border: node.parent != null
+                ? Border(left: border, bottom: border)
+                : null,
+          ),
+          child: const SizedBox(height: 50.0, width: 20.0),
+        ),
+        // Leading icon for parent nodes
+        if (isParentNode)
+          DecoratedBox(
+            decoration: BoxDecoration(border: Border.all()),
+            child: SizedBox.square(
+              dimension: 20.0,
+              child: Icon(
+                node.isExpanded ? Icons.remove : Icons.add,
+                size: 14,
+              ),
+            ),
+          ),
+        // Spacer
+        const SizedBox(width: 8.0),
+        // Content
+        Text(node.content.toString()),
+      ],
+    );
+  }
+
+  Map<Type, GestureRecognizerFactory> _getTapRecognizer(
+    TreeViewNode<String> node,
+  ) {
+    return <Type, GestureRecognizerFactory>{
+      TapGestureRecognizer:
+          GestureRecognizerFactoryWithHandlers<TapGestureRecognizer>(
+        () => TapGestureRecognizer(),
+        (TapGestureRecognizer t) => t.onTap = () {
+          setState(() {
+            // Toggling the node here instead means any tap on the row can
+            // toggle parent nodes opened and closed.
+            treeController.toggleNode(node);
+            _selectedNode = node;
+          });
+        },
+      ),
+    };
+  }
+
+  Widget _getTree() {
+    return DecoratedBox(
+      decoration: BoxDecoration(
+        border: Border.all(),
+      ),
+      child: Scrollbar(
+        controller: _horizontalController,
+        thumbVisibility: true,
+        child: Scrollbar(
+          controller: verticalController,
+          thumbVisibility: true,
+          child: TreeView<String>(
```

Comment:
Is there a way to customize the padding between the nodes and the labels? 
It looks like gap between the nodes and labels is bit too high when you see something like GitHub or VS Code tree list.

Preview 1  | Preview 2 | Preview 3 |
| --------------- | --------------- |--------------- |
| <img src="https://github.com/flutter/packages/assets/48603081/51022dbb-94b3-4f7d-be06-3a50a40c9cd4" > |  <img src="https://github.com/flutter/packages/assets/48603081/c5bed197-1f27-4b82-8c73-71582558e716" > | <img src="https://github.com/flutter/flutter/assets/48603081/fe6ade2a-280f-4fb2-9a46-d772748864c6" > |

Reply:
Yup! The padding there comes from the TreeView.defaultTreeNodeBuilder. Folks could copy it and remove the padding, or provide their own builder to TreeView.treeNodeBuilder.